### ISSUE TITLE
Override dumper app public_addr and add tags to it

### ISSUE THREAD

**Issue by Brunomachadob** [gravitational/teleport#36005]
## What would you like Teleport to do?

I would like the dumper app to have labels as well as having a configurable `public_addr` like other apps.

## What problem does this solve?

Just some UX improvements as well as better access control to the dumper app.

### Dumper app labels
Due to the fact the dumper app does not have any tags, I cannot make it available to certain roles, unless I configure the role with allow all:
```
  allow:
    app_labels:
      '*': '*'
```
_or maybe there is a way to filter by name I'm not aware of? I've tried `name: dumper` and `app_name: dumper`_

<img width="615" alt="image" src="https://github.com/gravitational/teleport/assets/4688763/e01db09e-cfb3-408f-a52f-67ad7f603897">

### Public URL
We override the `public_addr` of all apps so we can shorten it to `my-app.mydomain.com` instead of `my-app.teleport.mydomain.com`. 
On our setup we make use of an wildcard DNS record for all these, also with a wildcard cert. 
This means to use the dumper I had to create yet another DNS record as well as signing a cert specifically for `dumper.teleport.mydomain.com`.

**Comment by strideynet** [comment]
As a workaround, it'd be fairly trivial to host something similar to "dumper" e.g https://github.com/postmanlabs/httpbin

I'd be careful of relying on "dumper" behaving a certain way or existing forever - it's mostly there for debugging/experimental purposes.

**Comment by Brunomachadob** [comment]
Yep, that is a possibility, but a bit trickier as actually deploying a service on our infra is more complicated than just using the dumper 😅 

I also agree it is a temporary thing to be used from time to time, but when we get need to use it, then we need to allow full access to all things just to be able to see it. We usually don't have a admin that sees everything.
